core/src/test/resources/sql/sub-query.iq

```diff
@@ -1761,9 +1761,10 @@ select sal from "scott".emp e
 
 !ok
 EnumerableCalc(expr#0..4=[{inputs}], expr#5=[NOT($t4)], expr#6=[IS NOT NULL($t4)], expr#7=[OR($t5, $t6)], expr#8=[IS NOT TRUE($t7)], SAL=[$t1], $condition=[$t8])
-  EnumerableHashJoin(condition=[=($2, $3)], joinType=[left])
-    EnumerableCalc(expr#0..7=[{inputs}], EMPNO=[$t0], SAL=[$t5], DEPTNO=[$t7])
-      EnumerableTableScan(table=[[scott, EMP]])
+  EnumerableMergeJoin(condition=[=($2, $3)], joinType=[left])
```

Comment:
I am bit concerned about some of the "new" plans in `sub-query.iq`, which introduce `EnumerableMergeJoin` (with `joinType=left`), but only one of the join inputs is sorted with an `EnumerableSort` (in this example the left input). The other input (in this case, the right) does not contain the Sort operator, so this could be a problem.

Reply:
Agreed. In fact, I am actually thinking this is a case of wrong plan (more serious than a "concern"). Because for merge join implementation (not only Enumerables but other system's implementation), I think assuming input is sorted is reasonable. So for systems that only use Calcite as a planner, they might see surprising plans

Reply:
Top-down optimization could fix this problem thought (which is not enabled by default). The top-down opt can enforce collations by inserting Sort.